chatsky/pipeline/pipeline/utils.py

```diff
@@ -32,20 +29,10 @@ def rename_component_incrementing(component: PipelineComponent, collisions: List
     :param collisions: Components in the same service group as component.
     :return: Generated name
     """
-    if isinstance(component, Actor):
-        base_name = "actor"
-    # Pretty sure that service.handler can only be a callable now. Should the newly irrelevant logic be removed?
-    elif isinstance(component, Service) and callable(component.handler):
-        if isfunction(component.handler):
-            base_name = component.handler.__name__
-        else:
-            base_name = component.handler.__class__.__name__
-    elif isinstance(component, ServiceGroup):
-        base_name = "service_group"
+    if isinstance(component, PipelineComponent):
+        base_name = component.computed_name
     else:
-        # This should never be triggered. (All PipelineComponent derivatives are handled above)
         base_name = "noname_service"
```

Comment:
Now this definitely should never trigger.
Can either remove this if-else block or raise here.

Personally, I prefer removing the block:
- type of `component` can be validated with `validate_call`.
- it will raise if `component` does not have `computed_name`
- the block increases the number of lines